### PR TITLE
Topic cleanup continued

### PR DESCRIPTION
**Pull request by LorenzMeier**
Rebase on master once `topic_cleanup` is in. This removes additionally the attitude setpoint rotation matrix.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@tumbili @Stifael Can you both review this carefully? This again reduces the logging data rate considerably.

**Comment by LorenzMeier** [comment]
SITL tested including transitions.

File: src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp [hunk 1]
```diff
@@ -930,12 +932,14 @@ void  MulticopterPositionControlMultiplatform::handle_vehicle_attitude(const px4
 						_R(i, 2) = body_z(i);
 					}
 
-					/* copy rotation matrix to attitude setpoint topic */
-					memcpy(&_att_sp_msg.data().R_body[0], _R.data, sizeof(_att_sp_msg.data().R_body));
-					_att_sp_msg.data().R_valid = true;
-
 					/* calculate euler angles, for logging only, must not be used for control */
 					math::Vector<3> eul = _R.to_euler();
+
```
Comment (Stifael) [inline src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp]: move above euler angles computation such that calculation and assignment are one block

File: src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp [hunk 2]
```diff
@@ -930,12 +932,14 @@ void  MulticopterPositionControlMultiplatform::handle_vehicle_attitude(const px4
...
+					math::Quaternion q;
+					q.from_dcm(_R);
+					memcpy(_att_sp_msg.data().q_d, q.data, sizeof(_att_sp_msg.data().q_d));
+					_att_sp_msg.data().q_d_valid = true;
```
Comment (Stifael) [inline src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp]: move block above "calculate euler angles...", such that computation and assignment are together

File: src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp [hunk 3]
```diff
@@ -640,8 +640,10 @@ void  MulticopterPositionControlMultiplatform::handle_vehicle_attitude(const px4
 		    && _pos_sp_triplet->data().current.type == _pos_sp_triplet->data().current.SETPOINT_TYPE_IDLE) {
 			/* idle state, don't run controller and set zero thrust */
 			_R.identity();
-			memcpy(&_att_sp_msg.data().R_body[0], _R.data, sizeof(_att_sp_msg.data().R_body));
-			_att_sp_msg.data().R_valid = true;
+			math::Quaternion qd;
+			qd.from_dcm(_R);
+			memcpy(&_att_sp_msg.data().q_d[0], qd.data, sizeof(_att_sp_msg.data().q_d));
```
Comment (Stifael) [inline src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp]: Syntax cleanup: here  reference operator (&) used, but not necessary. suggestion:
memcpy(_att_sp_msg.data().q_d, qd.data, sizeof(_att_sp_msg.data().q_d))

File: src/modules/mavlink/mavlink_messages.cpp [hunk 4]
```diff
@@ -2584,7 +2584,11 @@ class MavlinkStreamAttitudeTarget : public MavlinkStream
 			mavlink_attitude_target_t msg{};
 
 			msg.time_boot_ms = att_sp.timestamp / 1000;
-			mavlink_euler_to_quaternion(att_sp.roll_body, att_sp.pitch_body, att_sp.yaw_body, msg.q);
+			if (att_sp.q_d_valid) {
```
Comment (Stifael) [inline src/modules/mavlink/mavlink_messages.cpp]: don't need & operator with 0 indexing

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 5]
```diff
@@ -1391,8 +1391,9 @@ MulticopterPositionControl::task_main()
 			    && _pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_IDLE) {
 				/* idle state, don't run controller and set zero thrust */
 				R.identity();
-				memcpy(&_att_sp.R_body[0], R.data, sizeof(_att_sp.R_body));
-				_att_sp.R_valid = true;
+				matrix::Quatf qd = R;
+				memcpy(&_att_sp.q_d[0], qd.data(), sizeof(_att_sp.q_d));
```
Comment (Stifael) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: dont need & and 0

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 6]
```diff
@@ -1651,14 +1650,16 @@ MulticopterPositionControl::task_main()
 					// if yes, then correct xy velocity setpoint such that the attitude setpoint is continuous
 					if (!control_vel_enabled_prev && _control_mode.flag_control_velocity_enabled) {
 
+						matrix::Dcmf Rb = matrix::Quatf(_att_sp.q_d[0], _att_sp.q_d[1], _att_sp.q_d[2], _att_sp.q_d[3]);
```
Comment (Stifael) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: @tumbili : he should know it that is correct

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 7]
```diff
@@ -1904,34 +1905,26 @@ MulticopterPositionControl::task_main()
...
-						/* copy rotation matrix to attitude setpoint topic */
-						memcpy(&_att_sp.R_body[0], R.data, sizeof(_att_sp.R_body));
-						_att_sp.R_valid = true;
-
 						/* copy quaternion setpoint to attitude setpoint topic */
-						math::Quaternion q_sp;
-						q_sp.from_dcm(R);
-						memcpy(&_att_sp.q_d[0], &q_sp.data[0], sizeof(_att_sp.q_d));
+						matrix::Quatf q_sp = R;
+						memcpy(&_att_sp.q_d[0], q_sp.data(), sizeof(_att_sp.q_d));
```
Comment (Stifael) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: don't need & and 0

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 8]
```diff
@@ -1904,34 +1905,26 @@ MulticopterPositionControl::task_main()
...
-						memcpy(&_att_sp.R_body[0], R.data, sizeof(_att_sp.R_body));
-						_att_sp.R_valid = true;
+						R = matrix::Eulerf(0.0f, 0.0f, _att_sp.yaw_body);
 
 						/* copy quaternion setpoint to attitude setpoint topic */
-						math::Quaternion q_sp;
-						q_sp.from_dcm(R);
-						memcpy(&_att_sp.q_d[0], &q_sp.data[0], sizeof(_att_sp.q_d));
+						matrix::Quatf q_sp = R;
+						memcpy(&_att_sp.q_d[0], q_sp.data(), sizeof(_att_sp.q_d));
```
Comment (Stifael) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: dont need & and 0

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 9]
```diff
@@ -2065,14 +2058,10 @@ MulticopterPositionControl::task_main()
...
-				math::Matrix<3, 3> R_sp;
-				R_sp.from_euler(_att_sp.roll_body, _att_sp.pitch_body, _att_sp.yaw_body);
-				memcpy(&_att_sp.R_body[0], R_sp.data, sizeof(_att_sp.R_body));
-
 				/* copy quaternion setpoint to attitude setpoint topic */
-				math::Quaternion q_sp;
-				q_sp.from_dcm(R_sp);
-				memcpy(&_att_sp.q_d[0], &q_sp.data[0], sizeof(_att_sp.q_d));
+				matrix::Quatf q_sp = matrix::Eulerf(_att_sp.roll_body, _att_sp.pitch_body, _att_sp.yaw_body);
+				memcpy(&_att_sp.q_d[0], q_sp.data(), sizeof(_att_sp.q_d));
```
Comment (Stifael) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: dont need & and 0

**Pull request merged**